packages/contracts-bedrock/test/L1/SystemConfigInterop.t.sol

```diff
@@ -0,0 +1,116 @@
+        vm.assume(bytes(_symbol).length <= 32);
+
+        vm.mockCall(_token, abi.encodeWithSelector(ERC20.decimals.selector), abi.encode(18));
+        vm.mockCall(_token, abi.encodeWithSelector(ERC20.name.selector), abi.encode(_name));
+        vm.mockCall(_token, abi.encodeWithSelector(ERC20.symbol.selector), abi.encode(_symbol));
+
+        vm.mockCall(
+            address(optimismPortal),
+            abi.encodeWithSelector(OptimismPortalInterop.setConfig.selector),
+            abi.encode(
+                ConfigType.GAS_PAYING_TOKEN,
+                StaticConfig.encodeSetGasPayingToken({
+                    _token: _token,
+                    _decimals: 18,
+                    _name: GasPayingToken.sanitize(_name),
+                    _symbol: GasPayingToken.sanitize(_symbol)
+                })
+            )
+        );
+
+        _systemConfigWithSetGasPayingToken().setGasPayingToken(_token);
+    }
+
+    /// @dev Tests that a dependency can be added.
+    function testFuzz_addDependency_succeeds(uint256 _chainId) public {
+        vm.mockCall(
+            address(optimismPortal),
+            abi.encodeWithSelector(OptimismPortalInterop.setConfig.selector),
+            abi.encode(ConfigType.GAS_PAYING_TOKEN, StaticConfig.encodeAddDependency(_chainId))
+        );
+
+        vm.prank(systemConfig.owner());
+        _systemConfigInterop().addDependency(_chainId);
+    }
+
+    /// @dev Tests that adding a dependency as not the owner reverts.
+    function testFuzz_addDependency_notOwner_reverts(uint256 _chainId) public {
+        vm.expectRevert("Ownable: caller is not the owner");
+        _systemConfigInterop().addDependency(_chainId);
+    }
+
+    /// @dev Tests that a dependency can be removed.
+    function testFuzz_removeDependency_succeeds(uint256 _chainId) public {
+        vm.mockCall(
+            address(optimismPortal),
+            abi.encodeWithSelector(OptimismPortalInterop.setConfig.selector),
+            abi.encode(ConfigType.REMOVE_DEPENDENCY, StaticConfig.encodeRemoveDependency(_chainId))
+        );
+
+        vm.prank(systemConfig.owner());
+        _systemConfigInterop().removeDependency(_chainId);
+    }
+
+    /// @dev Tests that removing a dependency as not the owner reverts.
+    function testFuzz_removeDependency_notOwner_reverts(uint256 _chainId) public {
+        vm.expectRevert("Ownable: caller is not the owner");
+        _systemConfigInterop().removeDependency(_chainId);
+    }
+
+    /// @dev Returns the SystemConfigInterop instance.
+    function _systemConfigInterop() internal view returns (SystemConfigInterop) {
+        return SystemConfigInterop(address(systemConfig));
+    }
+
+    /// @dev Returns the SystemConfigWithSetGasPayingToken instance.
+    function _systemConfigWithSetGasPayingToken() internal returns (SystemConfigWithSetGasPayingToken) {
+        vm.etch(address(systemConfig), address(new SystemConfigWithSetGasPayingToken()).code);
```

Comment:
Thoughts on the reinitialize approach used elsewhere in the tests? I would lean towards that instead of modifying bytecode, so we can have the exact bytecode we run in prod running

Reply:
good idea!